components/profile-controller/controllers/profile_controller.go

```diff
@@ -617,15 +630,48 @@ func removeString(slice []string, s string) (result []string) {
 	return
 }
 
+func enforceNamespaceLabelsFromConfig(ns *corev1.Namespace, logger logr.Logger) {
+	if ns.Labels == nil {
+		ns.Labels = make(map[string]string)
+	}
+
+	for k, v := range enforcedKubeflowNamespaceLabels {
+		existingValue, ok := ns.Labels[k]
+		if len(v) == 0 {
+			// When there is an empty value, k should be removed.
+			if ok {
+				delete(ns.Labels, k)
+			}
+		} else {
+			if !ok || existingValue != v {
```

Comment:
The if seems unnecessary, no matter what, you will set this label

Reply:
You are right! Removed this if condition.